### PR TITLE
ASM-7035 VSAN deployment fails with VSAN partition on back side disk

### PR DESCRIPTION
**Pull request by sushilrai**
Run import configuration twice to take care of removing backplane raid configuration. With 13G servers there is no way to initialize the disk in AHCI and ATA mode. For now we are deleting the VD when server is in Raidmode and then update the server from Raid mode to AHCI mode

### CODE REVIEW

**Comment by aimers1975** [comment]
@sushilrai is it expected that the tests are failing.. @asaenz are you able to review despite the failing tests.  Note this is an urgent 8.2.1 fix.

**Comment by sushilrai** [comment]
@ripienaar not sure which gem version has changed. Tests are failing due the version conflict which require ruby 2.0

**Comment by ripienaar** [comment]
@sushilrai looking, but yeah its not obvious whats going on...

File: lib/puppet/provider/importtemplatexml.rb [hunk 1]
```diff
@@ -311,8 +390,8 @@ def munge_config_xml
   def remove_invalid_settings(xml_to_edit)
     # HddSeq seems to cause a lot of issues by letting it stay.
     # We only allow it to stay if we're specifically trying to set it for booting from hard drive
-    hdd_seq = find_attribute_value(xml_to_edit, 'BIOS.Setup.1-1', 'HddSeq', false)
-    hdd_seq.remove unless hdd_seq.nil? || @changes['partial']['BIOS.Setup.1-1']['HddSeq']
+    #hdd_seq = find_attribute_value(xml_to_edit, 'BIOS.Setup.1-1', 'HddSeq', false)
```
Comment (asaenz) [inline lib/puppet/provider/importtemplatexml.rb]: I can't 100% remember the issues that these lines solved, but why is this being commented out?

File: lib/puppet/provider/importtemplatexml.rb [hunk 2]
```diff
@@ -19,11 +19,45 @@ def initialize (ip,username,password,resource, exported_postfix='base')
...
+        @import_operation_count += 1
+        export_config
+      end
+    else
+      munge_config_xml unless @xml_processed
+      execute_import
+    end
+  end
+
+  def export_config(name_postfix='original')
```
Comment (asaenz) [inline lib/puppet/provider/importtemplatexml.rb]: I don't like how much of this is copy/pasted from exporttemplatexml.rb. Couldn't you just use the exporttemplatexml object similar to here?  https://github.com/dell-asm/dell-idrac/blob/master/lib/puppet/provider/idrac.rb#L165

File: lib/puppet/provider/importtemplatexml.rb (hunk 1 above)
Comment (sushilrai) [inline lib/puppet/provider/importtemplatexml.rb]: "hdd_seq" is a string variable and string.remove is not a valid operation.

Comment (asaenz) [inline lib/puppet/provider/importtemplatexml.rb]: Oh. Wow, that's been in for quite a while. I wonder how that got past me...weird. Alright then.

**Comment by ripienaar** [comment]
@sushilrai i fixed the gem install thign

File: lib/puppet/provider/importtemplatexml.rb (hunk 1 above)
Comment (gavin-scott) [inline lib/puppet/provider/importtemplatexml.rb]: The issue they were trying to solve is the one where if you don't actually change the HddSeq or BiosBootSeq but leave it set to an existing value the idrac import config job hangs forever.

**Comment by gavin-scott** [comment]
@sushilrai do you think this PR is still needed? @aimers1975 reports it did not seem to solve the original problem.

**Comment by gavin-scott** [comment]
Closing, reopen if still needed.

**Pull request closed without merging**